File: src/main/java/by/forward/forward_system/core/dto/rest/authors/AuthorDisciplinesDto.java
package by.forward.forward_system.core.dto.rest.authors;

import by.forward.forward_system.core.dto.rest.DisciplineDto;
import lombok.Builder;
import lombok.Data;

import java.util.List;

@Data
@Builder
public class AuthorDisciplinesDto {
    private Long userId;
    private List<DisciplineDto> excellent;
    private List<DisciplineDto> good;
    private List<DisciplineDto> maybe;
}

File: forward-system-frontend/src/core/dto/AuthorDisciplinesDto.ts
import type {DisciplineDto} from "@/core/dto/DisciplineDto.ts";
import type {LastSeenAtDto} from "@/core/LastSeenAtService.ts";

export interface AuthorDisciplinesDto {
    userId: number,
    excellent: DisciplineDto[],
    good: DisciplineDto[],
    maybe: DisciplineDto[],
    activity: LastSeenAtDto,
    activeOrderIds: number[],
    maxOrdersCount: number
}